### PR TITLE
firestore: add samples for in and array-contains-any

### PR DESCRIPTION
**Pull request by tritone**
Add samples for these two query operators.

I compiled and ran locally to test.

### CODE REVIEW

File: firestore/firestore_snippets/query.go [hunk 1]
```diff
@@ -259,6 +269,16 @@ func createArrayContainsQuery(ctx context.Context, client *firestore.Client) err
 	return nil
 }
 
+func createArrayContainsAnyQuery(ctx context.Context, client *firestore.Client) error {
+	cities := client.Collection("cities")
+	// [START fs_array_contains_any_query]
```
Comment (BenWhitehead) [inline firestore/firestore_snippets/query.go]:
```suggestion
	// [START fs_query_filter_array_contains_any]
```

File: firestore/firestore_snippets/query.go [hunk 2]
```diff
@@ -259,6 +269,16 @@ func createArrayContainsQuery(ctx context.Context, client *firestore.Client) err
...
+	query := cities.Where("regions", "array-contains-any", []string{"west_coast", "east_coast"}).Documents(ctx)
+	// [END fs_array_contains_any_query]
```
Comment (BenWhitehead) [inline firestore/firestore_snippets/query.go]:
```suggestion
	// [END fs_query_filter_array_contains_any]
```

File: firestore/firestore_snippets/query.go [hunk 3]
```diff
@@ -249,6 +249,16 @@ func createMultipleStartAtQuery(client *firestore.Client) {
 	// [END fs_start_at_multiple]
 }
 
+func createInQuery(ctx context.Context, client *firestore.Client) error {
+	cities := client.Collection("cities")
+	// [START fs_in_query]
```
Comment (BenWhitehead) [inline firestore/firestore_snippets/query.go]:
```suggestion
	// [START fs_query_filter_in]
```

File: firestore/firestore_snippets/query.go [hunk 4]
```diff
@@ -249,6 +249,16 @@ func createMultipleStartAtQuery(client *firestore.Client) {
...
+	query := cities.Where("regions", "in", [][]string{{"west_coast"}, {"east_coast"}}).Documents(ctx)
+	// [END fs_in_query]
```
Comment (BenWhitehead) [inline firestore/firestore_snippets/query.go]:
```suggestion
	// [END fs_query_filter_in]
```

File: firestore/firestore_snippets/query.go [hunk 5]
```diff
@@ -249,6 +249,16 @@ func createMultipleStartAtQuery(client *firestore.Client) {
...
+
+	_ = query
+	return nil
+}
+
```
Comment (BenWhitehead) [inline firestore/firestore_snippets/query.go]: Can you also add a function for In query with an array of arrays? For example in java: https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/firestore/src/main/java/com/example/firestore/snippets/QueryDataSnippets.java#L464-L475 This scenario is specifically called out in the docs.

Comment (tritone) [reply firestore/firestore_snippets/query.go]: This is an array of arrays (I directly imitated the example in the docs here: https://firebase.google.com/docs/firestore/query-data/queries#in_and_array-contains-any )

Can also add one mirroring https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/firestore/src/main/java/com/example/firestore/snippets/QueryDataSnippets.java#L455 as well if you like.

Comment (BenWhitehead) [reply firestore/firestore_snippets/query.go]: Yeah, we want both scenarios. Sorry for the confusion.

File: firestore/firestore_snippets/query.go (hunk 3 above)
Comment (BenWhitehead) [reply firestore/firestore_snippets/query.go]: Can you also move this tag start above `cities := ...` We want that ref in the snippet.

File: firestore/firestore_snippets/query.go (hunk 1 above)
Comment (BenWhitehead) [reply firestore/firestore_snippets/query.go]: Can you move the start tag above the `cities := ...` line, we want that in the snippet.

Comment (tritone) [reply firestore/firestore_snippets/query.go]: The array-contains snippet in this file does not include that, but happy to include it in these.

**Pull request merged**